### PR TITLE
Adjust Makefiles for new generic prow jobs

### PR DESCRIPTION
**Pull request by kfurgol**
**Description**

Changes proposed in this pull request:

- Removed before-commit.sh scripts from the Compass components
- Adjusted Makefiles for new generic prow jobs

**Related issue(s)**
[#5746](https://github.com/kyma-project/kyma/issues/5746)

### CODE REVIEW

**Comment by dbadura** [comment]
\test pre-compass-tests-end-to-end

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+`go-dep-mk` is a generic make file used to build, format and test Golang components.
+GNU Makefile rule almost in every case contains 2 versions:
+* local -  this rule is not launched inside docker
+* not-local - this rule is launched inside docker with component directory mounted as volume 
+Possible checks for GNU Makefile:
+* check-fmt
+* check_imports
```
Comment (aszecowka) [inline scripts/README.md]: shouldn't be `check-imports`?
In addition to that, this README file does not help me to understand the current build process

Comment (dbadura) [reply scripts/README.md]: Aaaa README is under construction

File: tests/connector-tests/Makefile [hunk 2]
```diff
@@ -1,29 +1,17 @@
...
-ci-pr: resolve build-and-test build-image push-image
-ci-master: resolve build-and-test build-image push-image
-ci-release: resolve build-and-test build-image push-image
+VERIFY_IGNORE := /vendor\|/automock\|/testdata
+
+verify:: vet
+
+$(eval $(call test-local))
+
+#We have to override test-local, because
```
[the remark this answers is not in the document]
Comment (aszecowka) [reply tests/connector-tests/Makefile]: `test-local: ;` probably can be used

File: tests/end-to-end/director/testctx.go [hunk 3]
```diff
@@ -24,33 +23,17 @@ func init() {
...
-	err := scopeProvider.Load()
-	if err != nil {
-		return nil, errors.Wrap(err, "while loading config for scopes")
-	}
-
-	currentScopes, err := scopeProvider.GetAllScopes()
-	if err != nil {
-		return nil, err
-	}
+	scopes := os.Getenv("SCOPES_CONFIGURATION")
```
Comment (aszecowka) [inline tests/end-to-end/director/testctx.go]: 1. maybe rename it to ALL_POSSIBLE_SCOPES
2. document this env variable in README.md
3. inject this variable in the chart

Comment (dbadura) [reply tests/end-to-end/director/testctx.go]: `ALL_SCOPES` is good name.

File: tests/connector-tests/Makefile [hunk 4]
```diff
@@ -1,29 +1,17 @@
...
+include $(SCRIPTS_DIR)/go-dep.mk
 
-ci-pr: resolve build-and-test build-image push-image
-ci-master: resolve build-and-test build-image push-image
-ci-release: resolve build-and-test build-image push-image
+VERIFY_IGNORE := /vendor\|/automock\|/testdata
+
+verify:: vet
+
+$(eval $(call test-local))
```
[the remark this answers is not in the document]
Comment (dbadura) [reply tests/connector-tests/Makefile]: Removed

File: components/schema-migrator/Makefile [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
+
+format:
+	exit 0
+
+release:: build-image push-image
+
+build-image:
+	docker build -t $(IMG_NAME) .
 push-image:
-	docker tag $(IMG_NAME) $(IMG):$(TAG)
```
[the remark this answers is not in the document]
Comment (dbadura) [reply components/schema-migrator/Makefile]: leftovers

File: components/connector/Makefile [hunk 6]
```diff
@@ -1,29 +1,20 @@
...
+include $(SCRIPTS_DIR)/go-dep.mk
 
-ci-pr: resolve build-and-test build-image push-image
-ci-master: resolve build-and-test build-image push-image
-ci-release: resolve build-and-test build-image push-image
+VERIFY_IGNORE := /vendor\|/mocks\|/testdata
+
+verify:: gqlgen gqlgen-check
+
+gqlgen-check:
```
[the remark this answers is not in the document]
Comment (dbadura) [reply components/connector/Makefile]: moved `gqlgen-check` to generic makefile

File: components/schema-migrator/Makefile [hunk 7]
```diff
@@ -1,19 +1,28 @@
...
+APP_NAME = compass-schema-migrator
+APP_PATH = components/schema-migrator
+IMG_NAME = $(DOCKER_PUSH_REPOSITORY)$(DOCKER_PUSH_DIRECTORY)/$(APP_NAME)
+TAG := $(DOCKER_TAG)
+ENTRYPOINT = ./run.sh
+BUILDPACK = eu.gcr.io/kyma-project/prow/test-infra/bootstrap:v20181204-a6e79be
+VERIFY_IGNORE = /vendor\|/automock\|/testdata
 
-.PHONY: ci-pr ci-master ci-release build-image push-image
+.PHONY: verify format release
```
[the remark this answers is not in the document]
Comment (dbadura) [reply components/schema-migrator/Makefile]: Removed

File: scripts/README.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+- [Docker](https://www.docker.com/get-started)
+- [GNU Makefile](https://www.gnu.org/software/make/manual/make.html)
+
+## Usage
+Syntax:
+```bash
+make 'rule'
+```
+
+There are 2 possible environments to run those rules:
```
[the remark this answers is not in the document]
Comment (dbadura) [reply scripts/README.md]: Yes, `make verify`. It's inside README.

File: scripts/README.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+- if `resolve` pass, then the next rule will be executed.
+- if exit code of any rules is different than 0, makefile will abort execution and fail.
+
+Target types:
+- `MOUNT_TARGET` - mount component directory as volume to docker container
+- `COPY_TAGRT` - copy components files to docker container.
+
+## How to adjust makefile
+### How to disable current rule in local makefile
+In new Makefile add following line: `${rule}: ;`.
```
Comment (crabtree) [inline scripts/README.md]:
```suggestion
In new Makefile add the following line: `{rule}: ;`.
```

File: tests/end-to-end/Makefile [hunk 10]
```diff
@@ -1,30 +1,42 @@
...
+VERIFY_IGNORE := /vendor\|/automock\|/testdata
+
+verify:: test-e2e
+
+$(eval $(call test-local))
+
+# We have to override test-local, because we need to run director with database as docker containers and connected with custom network
+# and the container itsefl has to be connected to the network
+test-local:
+	@echo make test-local
```
Comment (crabtree) [inline tests/end-to-end/Makefile]: how about `test-local: ;`

File: tests/end-to-end/README.md [hunk 11]
```diff
@@ -4,7 +4,23 @@ Compass end-to-end consists of acceptance tests.
...
+| DIRECTOR_GRAPHQL_API        | "http://127.0.0.1:3000/graphql" | director graphql API URL                          |
+
+To run Director tests with running director and connector, execute:
 ```
 ./run.sh
-```
\ No newline at end of file
+```
+
+To run tests with Director and Database run:
```
Comment (crabtree) [inline tests/end-to-end/README.md]:
```suggestion
To run tests together with Director and Database run:
```

File: tests/provisioner-tests/Makefile [hunk 12]
```diff
@@ -1,29 +1,13 @@
...
-endif
-build-image: pull-licenses
-	docker build -t $(IMG_NAME) .
-push-image:
-	docker tag $(IMG_NAME) $(IMG):$(TAG)
-	docker push $(IMG):$(TAG)
\ No newline at end of file
+$(eval $(call test-local))
+
+#We have to override test-local
```
Comment (crabtree) [inline tests/provisioner-tests/Makefile]: comment not needed

File: tests/provisioner-tests/Makefile [hunk 13]
```diff
@@ -1,29 +1,17 @@
...
 
-.PHONY: ci-pr ci-master ci-release resolve build-and-test build-image push-image pull-licenses
+include $(SCRIPTS_DIR)/generic_make_go.mk
 
-ci-pr: resolve build-and-test build-image push-image
-ci-master: resolve build-and-test build-image push-image
-ci-release: resolve build-and-test build-image push-image
+verify:: vet
+
+$(eval $(call test-local))
```
Comment (Szymongib) [inline tests/provisioner-tests/Makefile]: Do we need this line?
From what I understand `test-local: ;` is enough to disable the `test-local` rule.

File: tests/connector-tests/Makefile [hunk 14]
```diff
@@ -1,29 +1,17 @@
...
-endif
-build-image: pull-licenses
-	docker build -t $(IMG_NAME) .
-push-image:
-	docker tag $(IMG_NAME) $(IMG):$(TAG)
-	docker push $(IMG):$(TAG)
\ No newline at end of file
+verify:: vet
+
+#We have to override test-local, because
```
Comment (Szymongib) [inline tests/connector-tests/Makefile]: I think we can remove this comment

**Pull request merged**